src/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs

```diff
@@ -3,19 +3,229 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.Net;
+using System.Threading.Tasks;
 
 using Xunit;
 
 namespace System.Net.Primitives.Unit.Tests
 {
     public class CookieContainerTest
     {
+        private const string u1 = ".url1.com"; // Basic domain
+        private const string u2 = "127.0.0.1"; // Local domain
+        private const string u3 = "url3.com"; // Basic domain without a leading dot
+        private static readonly Uri u4 = new Uri("http://url4.com"); // Basic uri
+        private static readonly Uri u5 = new Uri("http://url5.com/path"); // Set explicit path
+        private static readonly Uri u6 = new Uri("http://url6.com"); // No cookies should be added with this uri
+
+        private static readonly Cookie c1 = new Cookie("name1", "value", "", u1);
+        private static readonly Cookie c2 = new Cookie("name2", "value", "", u1) { Secure = true };
+        private static readonly Cookie c3 = new Cookie("name3", "value", "", u1) { Port = "\"80, 90, 100, 443\"" };
+        private static readonly Cookie c4 = new Cookie("name4", "value", "", u2);
+        private static readonly Cookie c5 = new Cookie("name5", "value", "/path", u2);
+        private static readonly Cookie c6 = new Cookie("name6", "value", "", "." + u3);
+        private static readonly Cookie c7 = new Cookie("name7", "value", "", "." + u3);
+        private static readonly Cookie c8 = new Cookie("name8", "value");
+        private static readonly Cookie c9 = new Cookie("name9", "value");
+        private static readonly Cookie c10 = new Cookie("name10", "value");
+        private static readonly Cookie c11 = new Cookie("name11", "value") { Port = "\"80, 90, 100\"" };
```

Comment:
On second thought, there are other formatting changes we need to do so, I'm going to defer that to another PR.